pages/experience.tsx

```diff
@@ -30,29 +24,56 @@ const LoadingView = (): JSX.Element => (
   </main>
 );
 
-function Experience({ id }): JSX.Element {
+function Experience({ userId, dataId, isApparatusId }): JSX.Element {
   const [loading, setLoading] = useState(true);
-  const [jsonFile, setJsonFile] = useState([]);
+  const [experienceData, setExperienceData] = useState<ExperienceData>();
+  // either apparatusID is provided or experience id but not both
+
+  const experience: ExperienceData = {
+    apparatusMetadata: { Paths: [], Data: [] },
+    apparatusId: "",
+    initializationData: { actionList: [] },
+  };
 
   React.useEffect(() => {
-    getApparatusFromCloud(id).then((responseJson) => {
-      setJsonFile(responseJson);
-      setLoading(false);
-    });
-  }, [id]);
+    if (isApparatusId === "true") {
+      getApparatusFromCloud(dataId).then((apparatusJson) => {
+        experience.apparatusId = apparatusJson.Id.Identifier;
+        experience.apparatusMetadata = apparatusJson.Metadata;
+        setExperienceData(experience);
+        setLoading(false);
+      });
+    } else {
+      getExperienceFromCloud(userId, dataId).then((experienceJson) => {
+        experience.initializationData.actionList = experienceJson.actionList;
+        getApparatusFromCloud(experienceJson.apparatusId).then(
+          (apparatusJson) => {
+            experience.apparatusId = apparatusJson.Id.Identifier;
```

Comment:
Trying to load an experience crashes and displays "TypeError: apparatusJson.Id is undefined."
Also 2 code smells that need to be fixed.

Reply:
It was caused by merging main into this branch which affected the experience json file fields names. Fixed now

Reply:
Can confirm that I can now load apparatuses without it crashing.